var/spack/repos/builtin/packages/nwchem/package.py

```diff
@@ -119,6 +128,19 @@ def install(self, spec, prefix):
             args.extend(["LIBFFTW3=%s" % fftw.ld_flags])
             args.extend(["FFTW3_INCLUDE={0}".format(spec["fftw-api"].prefix.include)])
 
+        if "+libxc" in spec:
+            args.extend(["LIBXC_LIB=%s" % spec["libxc"].libs.ld_flags])
+            args.extend(["LIBXC_INCLUDE=%s" % spec["libxc"].prefix.include])
```

Comment:
We're switching the core to use f-strings (https://docs.python.org/3/tutorial/inputoutput.html) and encouraging it's use in packages.

Reply:
changed most of %s to f-string; f-string with [] (used by accessing value in spec by key) is unacceptable, those are replaced with a .format() subroutine.

Reply:
@yizeyi18 what do you mean by "unacceptable"?

You can do (note che use of `"` and `'`)
```python
args.extend([f"LIBXC_LIB={spec['libxc'].libs.ld_flags}")
```
instead of
```python
args.extend([f"LIBXC_LIB={0}".format(spec["libxc"].libs.ld_flags)])
```

But I'm not sure which option is more readable.

Reply:
@RMeli Python interpreter said something like "unclosed '[' ", I could not remember. Maybe I used " in spec[] at that time?
I think both the two seems OK. Which do you, or spack developers prefer?

Reply:
The `SyntaxError: f-string: unmatched '['` is the error you get if you use `"` for both the outer and inner strings. If you use `"` for the outer string and `'` for the inner string (as shown above), it works. 
```python
d = {"a": 1}

s1 = f"{d["a"]}" # Fails
s2 = f"{d['a']}" # Works
```

I don't think it's important (unless @tldahlgren says otherwise), it was more FYI and to understand what you meant by "unacceptable".

Reply:
Eh, python interpreter reported some error at that time, and I thought "It's strange but it really fails. Python don't accept this syntax" so wrote down the "unacceptable". It is not something important, I support your point. Maybe it just failed as you shows.